### ISSUE TITLE
Windows private_key

### ISSUE THREAD

**Issue by freimer** [saltstack/salt#21055]
See internal issue 196.  Although I have private_key set in both the provider and the profile:

provider:
```yaml
ec2:
  id: 'use-instance-role-credentials'
  key: 'use-instance-role-credentials'
  keyname: 'redacted'
  private_key: /redacted
  ssh_interface: private_ips   
  ssh_username: ec2-user
  location: us-east-1
  provider: ec2
  minion:
    master: redacted
  tmp_dir: /tmp/.saltcloud/
  deploy_command: sh /tmp/.saltcloud/deploy.sh
  script: bootstrap-salt
  script_args: git 2015.2
```

profile:
```yaml
w12r2_micro_az1:
  provider: ec2
  image: ami-f298ea9a
  size: t2.micro
  subnetid: redacted
  securitygroupid:
    - redacted
  minion:
    master: redacted
  win_username: Administrator
  win_password: auto
  win_installer: /srv/salt/win/repo/salt-minion/Salt-Minion-2014.7.2-AMD64-Setup.exe
  userdata_file: /srv/salt/minion/windows-firewall.ps1
  private_key: redacted
```
I get this message when deploying an instance with this profile:
```
[DEBUG   ] Setting the tags returned: [{'resourceType': 'instance', 'resourceId': 'i-redacted', 'value': 'wintest3', 'key': 'Name'}]
[INFO    ] Created node wintest3
[INFO    ] Salt node data. Private_ip: redacted
[INFO    ] Salt interface set to: redacted
[DEBUG   ] MasterEvent PUB socket URI: ipc:///var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: ipc:///var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Sending event - data = {'_stamp': '2015-02-26T02:40:20.666401', 'ip_address': 'redacted', 'event': 'waiting for ssh'}
[DEBUG   ] Waiting for auto-generated Windows EC2 password
[ERROR   ] Failed to create VM wintest3. Configuration value 'private_key' needs to be set
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/cloud/__init__.py", line 1240, in create
    output = self.clouds[func](vm_)
  File "/usr/lib/python2.7/site-packages/salt/cloud/clouds/ec2.py", line 2198, in create
    vm_, data, ip_address, display_ssh_output
  File "/usr/lib/python2.7/site-packages/salt/cloud/clouds/ec2.py", line 1927, in wait_for_instance
    'key_file': vm_['private_key'],
KeyError: 'private_key'
Error: There was a profile error: Failed to deploy VM
```
Thanks!

**Comment by rallytime** [comment]
@freimer Thanks for the report! Can you post the output of `salt --versions-report`, please?

**Comment by rallytime** [comment]
@freimer Actually, did you submit a fix for this already in #21080?

**Comment by freimer** [comment]
Yes, I submitted the fix.

Sent from my iPhone

On Feb 26, 2015, at 1:41 PM, Nicole Thomas <notifications@github.com<mailto:notifications@github.com>> wrote:


@freimer<https://github.com/freimer> Actually, did you submit a fix for this already in #21080<https://github.com/saltstack/salt/pull/21080>?

**Comment by rallytime** [comment]
Great! Thanks so much!
